app/controllers/sales_controller.rb

```diff
@@ -0,0 +1,50 @@
+class SalesController < ApplicationController
+  def index
+    @title = "Sales"
```

Comment:
There is way too much happening in this one controller method. You should consider how to adjust by moving logic to another controller method or somewhere else, like the model.